tycho-baseline-plugin/src/main/java/org/eclipse/tycho/baseline/FeatureBaselineComparator.java

```diff
@@ -388,37 +384,28 @@ private static boolean propertyEquals(String baselineValue, String projectValue)
 		}
 		String normalizdBl = baselineValue.replaceAll("\\s", " ").replaceAll("\\s+", " ").trim();
 		String normalizdPr = projectValue.replaceAll("\\s", " ").replaceAll("\\s+", " ").trim();
-		if (normalizdBl.equals(normalizdPr)) {
-			return true;
-		}
-		return false;
+		return normalizdBl.equals(normalizdPr);
 	}
 
 	private static int indexOfDifference(CharSequence s1, CharSequence s2) {
-		int length = s1.length();
-		if (length < s2.length()) {
-			return indexOfDifference(s2, s1);
-		}
-		int i;
-		for (i = 0; i < length; i++) {
+		int length = Math.min(s1.length(), s2.length());
+		for (int i = 0; i < length; i++) {
 			char c1 = s1.charAt(i);
 			char c2 = s2.charAt(i);
 			if (c1 != c2) {
-				break;
+				return i;
 			}
 		}
-		return i;
-
+		return -1;
```

Comment:
I don't think this is correct, if a > b but there is no difference in the prefix it must then return the length.

Reply:
That's right. That was also the behavior before.
Changed it, thanks for pointing out.